client/client-lib/src/transaction.rs

```diff
@@ -52,6 +95,16 @@ impl TransactionBuilder {
             self.tx.signature = Some(signature);
         }
 
+        // move coins to pending state, awaiting a transaction
+        batch.append_from_iter(self.input_coins.iter().map(|(amount, coin)| {
+            BatchItem::delete(CoinKey {
+                amount,
+                nonce: coin.coin.0.clone(),
+            })
+        }));
+        batch.append_insert(PendingCoinsKey(self.tx.tx_hash()), self.input_coins);
+        batch.commit();
+
```

Comment:
Yeah you are right that could make sense because whenever we create coins (either as inputs or outputs) we want to ensure they are tied to a transaction so they don't get lost, that's why I moved coin input creation into the `TransactionBuilder` to enforce that through code.